The VCF files produced by the three workflows as well as the truthset (NA12878-Truthset.vcf) were filtered with chromosome 21 exome coordinates using VCFtools-0.1.13 (https://sourceforge.net/projects/vcftools/files/) with the following command:   

*vcftools –vcf PATH_TO_VCF_File–bed broadinstitute-human-exomechr21.bed --recode –keep-INFO-all*

**Process to generate Venn Diagrams:**

The final vcf files from previous step were compared to generate the venn diagram using python script (https://github.com/Slugger70/vcf-venn) with the following sample command:

*python vcf-venn.py -f PATH_TO_FILTERED_VCF_FILE_OF_GALAXY_WORKFLOW -f PATH_TO_FILTERED_VCF_FILE_OF_CPIPE_WORKFLOW –f PATH_TO_FILTERED_VCF_FILE_OF_CWL_WORKFLOW -f PATH_TO_FILTERED_VCF_FILE_OF_TRUTHSET -p OUTPUT_FILE_PREFIX -d OUTPUT_DIRECTORY_NAME*

The resultant text file venn-out.txt contains the summary of the overlapping variants in the input vcf files.

The following command can be used to get statistics for known and novel variants in the resultant vcf files generated by the *vcf-venn.py* script.
    
**Known variants:**
      
*for i in OUTPUT_FILE_PREFIX.*.vcf; do echo $i; cat $i | grep -v '^#' | cut -f3 | grep '^rs' | wc -l; done*
   
**Novel variants:**
   
*for i in NA12878_validation.*.vcf; do echo $i; cat $i | grep -v '^#' | cut -f3 | grep '\.' | wc -l; done*
